commitments/eigenda.go

```diff
@@ -29,12 +32,28 @@ func NewEigenDACommitment(input []byte, commitmentType EigenDACommitmentType) Ei
 	}
 }
 
+// NewEigenDACommitmentWithEncoding creates a new commitment with encoding from the given input.
+func NewEigenDACommitmentWithEncoding(input []byte, commitmentType EigenDACommitmentType,
+	encoding EncodingType) EigenDACommitment {
+	return EigenDACommitment{
+		prefix:   commitmentType,
+		b:        input,
+		encoding: &encoding,
+	}
+}
+
 // CommitmentType returns the commitment type of EigenDACommitment.
 func (c EigenDACommitment) CommitmentType() EigenDACommitmentType {
 	return c.prefix
 }
 
 // Encode adds a commitment type prefix self describing the commitment.
 func (c EigenDACommitment) Encode() []byte {
+	// For V2+ certificates with encoding type
+	if c.prefix >= CertV2 && c.encoding != nil {
+		return append([]byte{byte(c.prefix), byte(*c.encoding)}, c.b...)
+	}
```

Comment:
hmm don't we want to always add the encoding byte for v2+ certs? why allow encoding to be nil?

Reply:
04dba8c7e4368cfd65fdfc4b9bb95f480a8ec008